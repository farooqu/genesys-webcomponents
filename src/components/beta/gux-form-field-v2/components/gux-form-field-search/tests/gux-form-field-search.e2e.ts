import { E2EPage, newE2EPage } from '@stencil/core/testing';
import {
  newSparkE2EPage,
  a11yCheck
} from '../../../../../../../tests/e2eTestUtils';

const axeExclusions = [];

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();

  return page;
}

describe('gux-form-field-search-beta', () => {
  describe('#render', () => {
    describe('label-position', () => {
      [
        '',
        'label-position="above"',
        'label-position="beside"',
        'label-position="screenreader"'
      ].forEach((componentAttribute, index) => {
        const html = `
          <gux-form-field-search-beta ${componentAttribute}>
            <input slot="input" type="search" value="Sample search"/>
            <label slot="label">Label</label>
          </gux-form-field-search-beta>
        `;

        it(`should render component as expected (${index + 1})`, async () => {
          const page = await newNonrandomE2EPage({ html });
          const element = await page.find('gux-form-field-search-beta');
          const elementShadowDom = await element.find(
            'pierce/.gux-form-field-container'
          );

          expect(element.outerHTML).toMatchSnapshot();
          expect(elementShadowDom).toMatchSnapshot();
        });

        it(`should be accessible (${index + 1})`, async () => {
          const page = await newSparkE2EPage({ html });

          await a11yCheck(page, axeExclusions);
        });
      });
    });

    describe('input attributes', () => {
      ['', 'disabled', 'required'].forEach((inputAttribute, index) => {
        const html = `
          <gux-form-field-search-beta>
            <input slot="input" type="search" value="Sample search" ${inputAttribute}/>
            <label slot="label">Label</label>
          </gux-form-field-search-beta>
        `;

        it(`should render component as expected (${index + 1})`, async () => {
          const page = await newNonrandomE2EPage({ html });
          const element = await page.find('gux-form-field-search-beta');
          const elementShadowDom = await element.find(
            'pierce/.gux-form-field-container'
          );

          expect(element.outerHTML).toMatchSnapshot();
          expect(elementShadowDom).toMatchSnapshot();
        });

        it(`should be accessible (${index + 1})`, async () => {
          const page = await newSparkE2EPage({ html });

          await a11yCheck(page, axeExclusions);
        });
      });
    });
  });
});
