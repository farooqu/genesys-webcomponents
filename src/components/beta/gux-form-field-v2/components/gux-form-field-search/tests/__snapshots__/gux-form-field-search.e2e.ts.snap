// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-search-beta #render input attributes should render component as expected (1) 1`] = `"<gux-form-field-search-beta class=\\"hydrated\\"> <input slot=\\"input\\" type=\\"search\\" value=\\"Sample search\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-search-beta>"`;

exports[`gux-form-field-search-beta #render input attributes should render component as expected (1) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <gux-icon class="hydrated" icon-name="search"></gux-icon>
        <slot name="input"></slot>
        <gux-form-field-input-clear-button class="hydrated"></gux-form-field-input-clear-button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-search-beta #render input attributes should render component as expected (2) 1`] = `"<gux-form-field-search-beta class=\\"hydrated\\"> <input slot=\\"input\\" type=\\"search\\" value=\\"Sample search\\" disabled=\\"\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-search-beta>"`;

exports[`gux-form-field-search-beta #render input attributes should render component as expected (2) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-disabled gux-input-container">
        <gux-icon class="hydrated" icon-name="search"></gux-icon>
        <slot name="input"></slot>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-search-beta #render input attributes should render component as expected (3) 1`] = `"<gux-form-field-search-beta class=\\"hydrated\\"> <input slot=\\"input\\" type=\\"search\\" value=\\"Sample search\\" required=\\"\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-search-beta>"`;

exports[`gux-form-field-search-beta #render input attributes should render component as expected (3) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label gux-required">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <gux-icon class="hydrated" icon-name="search"></gux-icon>
        <slot name="input"></slot>
        <gux-form-field-input-clear-button class="hydrated"></gux-form-field-input-clear-button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-search-beta #render label-position should render component as expected (1) 1`] = `"<gux-form-field-search-beta class=\\"hydrated\\"> <input slot=\\"input\\" type=\\"search\\" value=\\"Sample search\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-search-beta>"`;

exports[`gux-form-field-search-beta #render label-position should render component as expected (1) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <gux-icon class="hydrated" icon-name="search"></gux-icon>
        <slot name="input"></slot>
        <gux-form-field-input-clear-button class="hydrated"></gux-form-field-input-clear-button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-search-beta #render label-position should render component as expected (2) 1`] = `"<gux-form-field-search-beta label-position=\\"above\\" class=\\"hydrated\\"> <input slot=\\"input\\" type=\\"search\\" value=\\"Sample search\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-search-beta>"`;

exports[`gux-form-field-search-beta #render label-position should render component as expected (2) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <gux-icon class="hydrated" icon-name="search"></gux-icon>
        <slot name="input"></slot>
        <gux-form-field-input-clear-button class="hydrated"></gux-form-field-input-clear-button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-search-beta #render label-position should render component as expected (3) 1`] = `"<gux-form-field-search-beta label-position=\\"beside\\" class=\\"hydrated\\"> <input slot=\\"input\\" type=\\"search\\" value=\\"Sample search\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-search-beta>"`;

exports[`gux-form-field-search-beta #render label-position should render component as expected (3) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <gux-icon class="hydrated" icon-name="search"></gux-icon>
        <slot name="input"></slot>
        <gux-form-field-input-clear-button class="hydrated"></gux-form-field-input-clear-button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-search-beta #render label-position should render component as expected (4) 1`] = `"<gux-form-field-search-beta label-position=\\"screenreader\\" class=\\"hydrated\\"> <input slot=\\"input\\" type=\\"search\\" value=\\"Sample search\\" id=\\"gux-form-field-input-i\\"> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-search-beta>"`;

exports[`gux-form-field-search-beta #render label-position should render component as expected (4) 2`] = `
<div class="gux-form-field-container gux-screenreader">
  <div class="gux-form-field-label gux-screenreader">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input">
      <div class="gux-input-container">
        <gux-icon class="hydrated" icon-name="search"></gux-icon>
        <slot name="input"></slot>
        <gux-form-field-input-clear-button class="hydrated"></gux-form-field-input-clear-button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon class="hydrated" icon-name="alert-warning-octogon"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
  </div>
</div>
`;
